digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/linearasset/PostGISLinearAssetDao.scala

```diff
@@ -694,6 +697,106 @@ class PostGISLinearAssetDao() {
+
+    MassQuery.executeBatch(statement) { ps =>
+      props.foreach { case (assetId, prop) =>
+        ps.setInt(1, prop.values.head.value.toString.toInt)
+        ps.setLong(2, assetId)
+        ps.setLong(3, propertyId)
+        ps.addBatch()
+      }
+    }
+  }
+
+  private def updateDatePeriodMass(assetTypeId: Int, publicId: String, props: Seq[(Long, DynamicProperty)]) = {
+    val propertyId = Q.query[(String, Int), Long](propertyIdByPublicIdAndTypeId).apply(publicId, assetTypeId).first
+    val statement = "update date_period_value set start_date = TO_TIMESTAMP(?, 'YYYY-MM-DD\"T\"HH24:MI:SS.FF3'), end_date = TO_TIMESTAMP(?, 'YYYY-MM-DD\"T\"HH24:MI:SS.FF3') where asset_id = (?) and property_id = (?)"
+
+    MassQuery.executeBatch(statement) { ps =>
+      props.foreach { case (assetId, prop) =>
+        val startDate = prop.values.head.value.asInstanceOf[Map[String, String]].get("startDate").get
+        val endDate = prop.values.head.value.asInstanceOf[Map[String, String]].get("endDate").get
+        ps.setString(1, DatePropertyFormat.parseDateTime(startDate).toString)
+        ps.setString(2, DatePropertyFormat.parseDateTime(endDate).toString)
+        ps.setLong(3, assetId)
+        ps.setLong(4, propertyId)
+        ps.addBatch()
+      }
+    }
+  }
+
+  private def updateMultipleChoiceMass(assetTypeId: Int, publicId: String, props: Seq[(Long, DynamicProperty)]): Unit = {
+    val propertyId = Q.query[(String, Int), Long](propertyIdByPublicIdAndTypeId).apply(publicId, assetTypeId).first
+    val statement =
+      "update multiple_choice_value set enumerated_value_id = (select id from enumerated_value where value = (?) and property_id = (?)) where asset_id = (?) and property_id = (?)"
+
+    MassQuery.executeBatch(statement) { ps =>
+      props.foreach { case (assetId, prop) =>
+        ps.setInt(1, prop.values.head.value.toString.toInt)
+        ps.setLong(2, propertyId)
+        ps.setLong(3, assetId)
+        ps.setLong(4, propertyId)
+        ps.addBatch()
+      }
+    }
+  }
+
+  /**
+   * Used to update value adjustments from LinearAssetUpdater.updateChangeSet. The updates supported are number property values
+   * (use case: road width) and date period and checkbox (use case: annual repetition of damaged by thaw period)
+   * @param adjustments Value adjustments from LinearAssetUpdater
+   */
+  def updateValueAdjustments(adjustments: Seq[ValueAdjustment]): Unit = {
+    val assetTypeId = adjustments.head.assetTypeId
+    val properties = adjustments.foldLeft(Seq.empty[(Long, DynamicProperty)]) { (acc, adjustment) =>
+      acc ++ adjustment.value.asInstanceOf[DynamicValue].value.properties.map(prop =>
+        (adjustment.assetId, prop)
+      )
+    }.groupBy { case (_, dynamicProperty) =>
+      (dynamicProperty.propertyType, dynamicProperty.publicId)
+    }
+    properties.foreach { case ((propertyType, publicId), propsByAssetId) =>
```

Comment:
Milläs tavalla tää nyt tekee tämän massa operaation? 

properties.foreach taitaa käydä tän nyt vaan yksitellen?

Reply:
Sen pitäis kyl käydä ne keyt eli esim. leveydellä kaikki (integer, width) tulee samaan massaoperaatioon. Noi dynaamiset on pakko eritellä, koska niitä menee eri tauluihin. Kuitenkin käytännössä sieltä leveydeltä pitäisi tulla vain toi leveysarvo, joka menee tosta keyn perusteella samaan massaoperaatioon.